pkg/sql/sem/builtins/aggregate_builtins.go

```diff
@@ -114,7 +114,8 @@ var aggregates = map[string]builtinDefinition{
 					return types.MakeArray(args[0].ResolvedType())
 				},
 				newArrayAggregate,
-				"Aggregates the selected values into an array.",
+				"Aggregates the selected values into an array.\n"+
+				"This aggregate function is [non-commutative](#non-commutative).",
```

Comment:
I don't think it's a good idea to include markup in the description text. This makes the output of `\hf` in the interactive shell hard to read. I think the initial approach you cooked with David (a separate boolean) was better.